src/set/set.service.ts

```diff
@@ -1,345 +1,418 @@
+                    whenNotMatched: 'discard',
+                },
+            },
+        ])
+        // Since the aggregation has no return value, We have to make another call to get the updated data
+        const result = await this.setSchema.findById(setId, {
+            _id: 1,
+            truthCount: 1,
+            daresCount: 1,
+        })
+        /* istanbul ignore next */ // Unable to test Internal server error here
+        if (!result) throw new InternalServerErrorException()
 
-    // Recounts the active truths and dares, projects the new counts and merges them back into the existing document
-    const update = await this.setSchema.aggregate([
-      {
-        '$match': {
-          '_id': new Types.ObjectId(setId.toString())
+        return {
+            _id: result._id,
+            truthCount: result.truthCount,
+            daresCount: result.daresCount,
         }
-      }, {
-        '$set': {
-          'daresCount': {
-            '$size': {
-              '$filter': {
-                'input': '$tasks',
-                'as': 'a',
-                'cond': {
-                  '$and': [
-                    {
-                      '$eq': [
-                        '$$a.type', 'dare'
-                      ]
-                    }, {
-                      '$eq': [
-                        '$$a.status', 'active'
-                      ]
-                    }
-                  ]
-                }
-              }
-            }
-          },
-          'truthCount': {
-            '$size': {
-              '$filter': {
-                'input': '$tasks',
-                'as': 'a',
-                'cond': {
-                  '$and': [
+    }
+
+    // Migrations / Seeder
+    /* istanbul ignore next */ // This is development only
+    public async createExampleSets(user: JwtUserDto, test: string) {
+        SetSampleData.forEach(async (setData) => {
+            const set = await this.createSet(
+                {
+                    name: setData.name,
+                    language: setData.language,
+                },
+                user,
+            )
+            setData.tasks.forEach(async (task) => {
+                await this.createTask(
+                    set._id,
                     {
-                      '$eq': [
-                        '$$a.type', 'truth'
-                      ]
-                    }, {
-                      '$eq': [
-                        '$$a.status', 'active'
-                      ]
-                    }
-                  ]
-                }
-              }
-            }
-          }
+                        type: task.type,
+                        currentPlayerGender: task.currentPlayerGender,
+                        message: task.message,
+                    },
+                    user,
+                )
+            })
+        })
+        // TODO: This is a security breach due to the mock data and the need to clean it in testing
```

Comment:
still valid after adding env guard?

Reply:
Can be deleted on envGuard implementation, added TODO envGuard to find it later

Reply:
D